sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionAsyncDatabase.java

```diff
@@ -125,12 +118,9 @@ public CosmosAsyncClientEncryptionKey getClientEncryptionKey(String id) {
             throw new IllegalArgumentException("clientEncryptionKeyId is null or empty");
         }
 
-        EncryptionKeyWrapProvider encryptionKeyWrapProvider =
-            this.cosmosEncryptionAsyncClient.getEncryptionKeyWrapProvider();
-
-        if (!encryptionKeyWrapProvider.getProviderName().equals(newEncryptionKeyWrapMetadata.getType())) {
+        if (!this.cosmosEncryptionAsyncClient.getKeyEncryptionKeyResolverName().equals(newEncryptionKeyWrapMetadata.getType())) {
```

Comment:
seems can extract this logic out into a different method, since it has been used in multiple places.

Reply:
this is used in this file twice with just getter , so we should be good here for code readability